docs/yaml/functions/_build_target_base.yaml

```diff
@@ -328,3 +328,12 @@ kwargs:
 
       This allows renaming similar to the dependency renaming feature of cargo
       or `extern crate foo as bar` inside rust code.
+
+  rename:
+    type: list[str]
+    since: 1.6.99
+    description: |
+      If specified renames each output into corresponding file from
+      `rename` list during install.  Nested paths are allowed and they
+      are joined with `install_dir`. Length of `rename` list must be
+      equal to the number of outputs.
```

Comment:
That does not match the `[0, 1, len(outputs)]` in the code below, though I think what you documented is the behavior that is more desirable. Allowing `rename: []` as "no rename" is a bit too magic, and I would disallow it as you documented here.

Reply:
Good plan. That also made me clean up the rename type to be `(ContainerTypeInfo(list,str), NoneType)` everywhere so that I could set the default values to `None` instead of `[]`.

Reply:
Hrm. There isn't a way to generate a None value from the meson script though? I think I want to allow an empty list to disable renaming (as the existing rename parameters do).

Reply:
My plan is to eventually have a `default()` function that returns an opaque value that can be used for this purpose, but I really need to get done with the "fix all of the interpreter/build layering violations" and port the rest of the Meson codebase to use the typed argument helpers, which would make this all just magically happen...

For now I'm fine with "empty list means default behavior", as we've generally used it for that purpose in Meson, even if it's not intuitive at all.

Reply:
Thanks. I've even documented it as the value to use to elicit the default behavior.